src/spares/OpenCollar - shocker.lsl

```diff
@@ -78,69 +87,64 @@ Debug(string sStr) {
-        if (sName != "" && llGetSubString(sName, 0, 0) != "~") lAnimList += [sName];
-    }
-    string sText = "Current punishment animation is: "+g_sShockAnim+"\n\n";
-    sText += "Select a new animation to use as a punishment.\n\n";
-    g_kAnimDialog = Dialog(kID, sText, lAnimList, [UPMENU],0, iAuth);
+    list lAnims ;
+    if (g_iDefaultAnim) lAnims = [DEFAULT];
+    lAnims += g_lAnims;
+    string sText = "\nCurrent punishment animation is: "+g_sShockAnim+"\n\n";
+    sText += "Chose new animation to use as a punishment:\n";
+    Dialog(kID, sText, lAnims, [UPMENU],0, iAuth,"anim");
 }
 
-
 DialogSelectSound(key kID, integer iAuth)
 {
     list lSoundList = [DEFAULT];
     integer iMax = llGetInventoryNumber(INVENTORY_SOUND);
     integer i;
     string sName;
-    for (i=0;i<iMax;i++)
+    for (i = 0; i < iMax; i++)
     {
-        sName=llGetInventoryName(INVENTORY_SOUND, i);
+        sName = llGetInventoryName(INVENTORY_SOUND, i);
         //check here if the anim start with ~ or for some reason does not get a name returned (spares to check that all again in the menu ;)
         if (sName != "" && llGetSubString(sName, 0, 0) != "~") lSoundList += [sName];
     }
     lSoundList+=[g_sNoSound];
-    string sText = "Current sound is: "+g_sShockSound+"\n\n";
-    sText += "Select a new sound to use.\n\n";
-    g_kSoundDialog = Dialog(kID, sText, lSoundList, [UPMENU],0, iAuth);
+    string sText = "\nCurrent sound is: "+g_sShockSound+"\n\n";
+    sText += "Chose new sound to use:\n";
+    Dialog(kID, sText, lSoundList, [UPMENU],0, iAuth,"sound");
 }
 
 DialogHelp(key kID, integer iAuth)
 {
-    string sMessage = "Usage of Shocker.\n";
-    sMessage += "Put in front of each command your subs prefix then use them as followed:\n";
-    sMessage += "shock <seconds> where <seconds> is time in seconds to punish you pet.\n";
-    sMessage += "shock 0/stop/off ,  is stop to punish you pet immediately.\n";
-    sMessage += "shockanim <anim name> , make sure the animation is inside the collar.\n";
-    sMessage += "shocksound <sound name> , make sure the sound is inside the collar.";
-
-    g_kDialog = Dialog(kID, sMessage, ["Ok"], [], 0, iAuth);
-    Notify(kID, sMessage, FALSE) ;
+    string sMsg = "Shocker chat commands:\n\n";
+    sMsg += "%PREFIX% shocker <seconds>\n     where <seconds> is time in seconds to punish you pet.\n\n";
+    sMsg += "%PREFIX% shocker 0/stop/off\n     is stop to punish you pet immediately.\n\n";
+    sMsg += "%PREFIX% shocker animation <anim name> \n    make sure the animation is inside the %DEVICETYPE%.\n\n";
+    sMsg += "%PREFIX% shocker sound <sound name>\n    make sure the sound is inside the %DEVICETYPE%.\n";
+
+    //Dialog(kID, sMsg, ["Ok"], [], 0, iAuth,"help");
+    //llMessageLinked(LINK_ROOT, NOTIFY, "0"+sMsg, kID);
+    llMessageLinked(LINK_ROOT, POPUP_HELP, sMsg, kID);
```

Comment:
llMessageLinked(LINK_ROOT,NOTIFY,"0"+sMsg,kID);
is the correct way here, the POPUP_HELP will be removed and does not handle %PREFIX% or %DEVICETYPE% but only the old _PREFIX_ and _CHANNEL_ where NOTIFY handles:
%PREFIX%, %WEARERNAME%, %CHANNEL% and %DEVICETYPE% plus %NOACCESS%